File: quivi.backoffice.react.webapp/src/hooks/api/Dtos/menuCategories/MenuCategory.ts
import { Language } from "../Language";

export interface MenuCategory {
    readonly id: string;
    readonly name: string;
    readonly imageUrl?: string;
    readonly sortIndex: number;
    readonly translations: Record<Language, MenuCategoryTranslation>;
}

export interface MenuCategoryTranslation {
    readonly name: string;
}

File: Quivi.Backoffice.Api/Dtos/MenuCategory.cs
using Quivi.Domain.Entities;

namespace Quivi.Backoffice.Api.Dtos
{
    public class MenuCategory
    {
        public required string Id { get; init; }
        public required string Name { get; init; }
        public string? ImageUrl { get; init; }
        public int SortIndex { get; init; }
        public required IReadOnlyDictionary<Language, MenuCategoryTranslation> Translations { get; init; }
    }

    public class MenuCategoryTranslation
    {
        public required string Name { get; init; }
    }
}
